File: University/lib/Dto/Zystem/ZUG2Dto.dart

class ZUG2Dto
{
	String ZHCONO = "";
	String ZHBRNO = "";
	String ZHUGNO = "";
	String ZHUSNO = "";
	String ZHSYST = "";
	String ZHSTAT = "";
	double ZHRCST = 0;
	double ZHCRDT = 0;
	double ZHCRTM = 0;
	String ZHCRUS = "";
	double ZHCHDT = 0;
	double ZHCHTM = 0;
	String ZHCHUS = "";
	String ZUUSNA = "";
	String ZUNICK = "";

	int PageNumber = 0;
	int PageSize = 0;
	int TotalPage = 0;
	int TotalRecord = 0;
	bool IsSelected = false;
	String SqlFilter = "";
	String SqlSort = "";

	ZUG2Dto({
		this.ZHCONO = "",
		this.ZHBRNO = "",
		this.ZHUGNO = "",
		this.ZHUSNO = "",
		this.ZHSYST = "",
		this.ZHSTAT = "",
		this.ZHRCST = 0,
		this.ZHCRDT = 0,
		this.ZHCRTM = 0,
		this.ZHCRUS = "",
		this.ZHCHDT = 0,
		this.ZHCHTM = 0,
		this.ZHCHUS = "",
		this.ZUUSNA = "",
		this.ZUNICK = "",

		this.PageNumber = 0,
		this.PageSize = 0,
		this.TotalPage = 0,
		this.TotalRecord = 0,
		this.IsSelected = false,
		this.SqlFilter = "",
		this.SqlSort = "",
	});

	factory ZUG2Dto.fromJson(Map<String, dynamic> json) {
		return ZUG2Dto(
			ZHCONO: json["ZHCONO"] ?? "",
			ZHBRNO: json["ZHBRNO"] ?? "",
			ZHUGNO: json["ZHUGNO"] ?? "",
			ZHUSNO: json["ZHUSNO"] ?? "",
			ZHSYST: json["ZHSYST"] ?? "",
			ZHSTAT: json["ZHSTAT"] ?? "",
			ZHRCST: json["ZHRCST"] ?? 0,
			ZHCRDT: json["ZHCRDT"] ?? 0,
			ZHCRTM: json["ZHCRTM"] ?? 0,
			ZHCRUS: json["ZHCRUS"] ?? "",
			ZHCHDT: json["ZHCHDT"] ?? 0,
			ZHCHTM: json["ZHCHTM"] ?? 0,
			ZHCHUS: json["ZHCHUS"] ?? "",
			ZUUSNA: json["ZUUSNA"] ?? "",
			ZUNICK: json["ZUNICK"] ?? "",

			PageNumber: json["PageNumber"] ?? 0,
			PageSize: json["PageSize"] ?? 0,
			TotalPage: json["TotalPage"] ?? 0,
			TotalRecord: json["TotalRecord"] ?? 0,
			IsSelected: json["IsSelected"] ?? false,
			SqlFilter: json["SqlFilter"] ?? "",
			SqlSort: json["SqlSort"] ?? "",
		);
	}

	Map toMap()
	{
		var map = <String, dynamic>{};
		map["ZHCONO"] = ZHCONO;
		map["ZHBRNO"] = ZHBRNO;
		map["ZHUGNO"] = ZHUGNO;
		map["ZHUSNO"] = ZHUSNO;
		map["ZHSYST"] = ZHSYST;
		map["ZHSTAT"] = ZHSTAT;
		map["ZHRCST"] = ZHRCST;
		map["ZHCRDT"] = ZHCRDT;
		map["ZHCRTM"] = ZHCRTM;
		map["ZHCRUS"] = ZHCRUS;
		map["ZHCHDT"] = ZHCHDT;
		map["ZHCHTM"] = ZHCHTM;
		map["ZHCHUS"] = ZHCHUS;
		map["ZUUSNA"] = ZUUSNA;
		map["ZUNICK"] = ZUNICK;

		map["PageNumber"] = PageNumber;
		map["PageSize"] = PageSize;
		map["TotalPage"] = TotalPage;
		map["TotalRecord"] = TotalRecord;
		map["IsSelected"] = IsSelected;
		map["SqlFilter"] = SqlFilter;
		map["SqlSort"] = SqlSort;
		return map;
	}
}

File: University.Service/University.Dto/Zystem/ZUG2Dto.cs
using System;
using System.Data;
using System.Collections.Generic;
using System.Runtime.Serialization;

using University.Dto.Base;

namespace University.Dto.Zystem
{
    [DataContract]
    public class ZUG2Dto
    {
        #region General Property

        [DataMember]
        public string ZHCONO { get; set; }

        [DataMember]
        public string ZHBRNO { get; set; }

        [DataMember]
        public string ZHUGNO { get; set; }

        [DataMember]
        public string ZHUSNO { get; set; }

        [DataMember]
        public string ZHSYST { get; set; }

        [DataMember]
        public string ZHSTAT { get; set; }

        [DataMember]
        public decimal ZHRCST { get; set; }

        [DataMember]
        public decimal ZHCRDT { get; set; }

        [DataMember]
        public decimal ZHCRTM { get; set; }

        [DataMember]
        public string ZHCRUS { get; set; }

        [DataMember]
        public decimal ZHCHDT { get; set; }

        [DataMember]
        public decimal ZHCHTM { get; set; }

        [DataMember]
        public string ZHCHUS { get; set; }


        #endregion        

        #region Custom Property

        [DataMember]
        public string ZUUSNA { get; set; }

        [DataMember]
        public string ZUNICK { get; set; }

        [DataMember]
        public decimal ZURCST { get; set; }

        [DataMember]
        public string ZRVANA { get; set; }

        [DataMember]
        public List<ZUSRDto> listZUSR { get; set; }

        [DataMember]
        public ZUSRDto objZUSR { get; set; }

        [DataMember]
        public string Result { get; set; }

        [DataMember]
        public int PageNumber { get; set; }

        [DataMember]
        public int PageSize { get; set; }

        [DataMember]
        public int TotalPage { get; set; }

        [DataMember]
        public int TotalRecord { get; set; }

        [DataMember]
        public bool IsSelected { get; set; }

        [DataMember]
        public string SqlFilter { get; set; }

        [DataMember]
        public string SqlSort { get; set; }
        #endregion
    }

    public class ZUG2MappingDto : Mapper<ZUG2Dto>
    {
        protected override ZUG2Dto PopulateItem(IDataRecord dr)
        {
            ZUG2Dto dto = new ZUG2Dto();
            for (int i = 0; i < dr.FieldCount; i++)
            {
                MapProperty(dto, dr.GetName(i), dr[i]);
            }
            return dto;
        }
    }
}
